File: example/main_button.py
#MODEL
model = {'num' : 0}

#VIEW
def on_forever():
    
    #UPDATE
    def onPressButton_A():
        global model
        model['num'] = model['num'] + 1
    input.on_button_pressed(Button.A, onPressButton_A)
    
    #UPDATE
    def onPressButton_B():
        global model
        model['num'] = model['num'] - 1
    input.on_button_pressed(Button.B, onPressButton_B)
    
    basic.show_number(model['num'])
    
basic.forever(on_forever)

File: example/main_button.ts
// MODEL
let model = {
    "num" : 0,
}

// VIEW
basic.forever(function on_forever() {
    // UPDATE
    input.onButtonPressed(Button.A, function onPressButton_A() {
        
        model["num"] = model["num"] + 1
    })
    // UPDATE
    input.onButtonPressed(Button.B, function onPressButton_B() {
        
        model["num"] = model["num"] - 1
    })
    basic.showNumber(model["num"])
})
